src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -138,6 +141,36 @@ int CPUPerformanceInterface::CPUPerformance::cpu_load_total_process(double* cpu_
 
   *cpu_load = (double)used_delta / total_delta;
 
+  return OS_OK;
+#elif defined(__FreeBSD__)
+  // counters are user/nice/system/interrupt/idle/states
+  long cpu_load_info[6];
+  size_t length = sizeof(long) * 6;
+
+  if (sysctlbyname("kern.cp_time", &cpu_load_info, &length, NULL, 0) == -1) {
+    return OS_ERR;
+  }
+
+  long used_ticks = cpu_load_info[0] + cpu_load_info[1] + cpu_load_info[2];
+  long total_ticks = used_ticks + cpu_load_info[4];
+
```

Comment:
Could you use the CP_USER, CP_NICE, etc defines instead of fixed values here as well?

I found this example on their use on FreeBSD:
https://github.com/freebsd/freebsd/blob/master/usr.bin/ctlstat/ctlstat.c#L210

Reply:
Good points sounds better indeed.